pkg/daemon/certs/certs.go

```diff
@@ -25,27 +25,103 @@ import (
 	"encoding/pem"
 	"errors"
 	"fmt"
+	"io"
 	"os"
+	"reflect"
 	"syscall"
 	"unsafe"
 
 	"golang.org/x/sys/windows"
 	"k8s.io/klog/v2"
 )
 
-// Reconcile ensures the certificates in the given trusted CA bundle file are imported as system certificates.
+// importedCABundleFile tracks all the certs that the operator has imported into the node's local trust store
+const importedCABundleFile = "C:\\Temp\\imported-certs.pem"
+
+// Reconcile ensures the node's system certificates match the expected state as given by the trusted CA bundle file.
 // Returns a boolean whether any certs needed to be reconciled
-func Reconcile(caBundle string) (bool, error) {
-	if caBundle == "" {
-		// TODO: ensure that all certs that we've added previously are removed https://issues.redhat.com/browse/WINC-688
+func Reconcile(caBundlePath string) (bool, error) {
+	// caBundlePath will be empty when a cluster-wide proxy is being removed or not in use
+	certChange, err := reconcileCerts(caBundlePath)
+	if err != nil {
+		return false, err
+	}
+	// An error updating the imported bundle file will be self-corrected on the next reconcile, since it is the source
+	// of truth for determining whether certs need to be updated
+	return certChange, updateImportedCABundle(caBundlePath, certChange)
+}
+
+// reconcileCerts reconciles any discrepency between expected and existing Windows certificates by importing or
+// deleting certificates from the root system store. Returns a boolean if any certificates were imported or deleted
+func reconcileCerts(caBundlePath string) (bool, error) {
+	expectedCerts, err := getExpectedCerts(caBundlePath)
+	if err != nil {
+		return false, err
+	}
+	existingCerts, err := getExistingCerts()
+	if err != nil {
+		return false, err
+	}
+
+	if reflect.DeepEqual(expectedCerts, existingCerts) {
 		return false, nil
 	}
-	// Read expected certs from CA trust bundle file
-	expectedCerts, err := readCertsFromFile(caBundle)
+
+	// Open the root certificate store
+	systemStore, err := windows.CertOpenStore(windows.CERT_STORE_PROV_SYSTEM, 0, 0,
+		windows.CERT_SYSTEM_STORE_LOCAL_MACHINE, uintptr(unsafe.Pointer(syscall.StringToUTF16Ptr("ROOT"))))
+	if err != nil {
+		return false, fmt.Errorf("Failed to open root certificate store: %w", err)
+	}
+	defer func() {
+		if err := windows.CertCloseStore(systemStore, 0); err != nil {
+			klog.Error("Failed to close root system certificate store")
+		}
+	}()
+
+	certsImported, err := ensureCertsAreImported(systemStore, expectedCerts, existingCerts)
+	if err != nil {
+		return false, fmt.Errorf("Failed to import certificates: %w", err)
+	}
+	certsDeleted, err := ensureCertsAreDeleted(systemStore, expectedCerts, existingCerts)
+	if err != nil {
+		return false, fmt.Errorf("Failed to delete certificates: %w", err)
+	}
```

Comment:
This block makes me think there should be a `CertManager` struct.
Specifically the passing in of `systemStore` to a bunch of functions.
Not a necessity, but could help organize the code a bit.

Reply:
to be clear you don't have to do this, just a nice to have